src/nvim/terminal.c

```diff
@@ -353,8 +353,14 @@ void terminal_resize(Terminal *term, uint16_t width, uint16_t height)
   invalidate_terminal(term, -1, -1);
 }
 
-void terminal_enter(Terminal *term, bool process_deferred)
+void terminal_enter(bool process_deferred)
 {
+  Terminal *term = curbuf->terminal;
+  assert(term && "should only be called when curbuf has a terminal");
```

Comment:
I did it like this because, if I understand the event loop correctly, we have to exit when `curbuf->terminal != term` or `term->buf != curbuf`, so even if `terminal_enter()` takes a `buf_T` argument, we'd have to assert `buf == curbuf && buf->terminal`.